dbms/src/Functions/FunctionsJSON.h

```diff
@@ -292,11 +293,25 @@ struct NameJSONExtractKeysAndValues { static constexpr auto name{"JSONExtractKey
 struct NameJSONExtractRaw { static constexpr auto name{"JSONExtractRaw"}; };
 
 
-template <typename JSONParser>
-class JSONHasImpl
+template <typename JSONParser, bool support_key_lookup>
```

Comment:
It looks a bit too complicated for this simple task. I suppose separate templates would be clearer:

```
template <typename JSONParser>
class JSONHasImpl
{
};

template <typename JSONParser>
class IsValidJSONImpl
{
};
```

Reply:
I create separate class, but reuse `JSONHasImpl<JSONParser>::addValueToColumn` in `addValueToColumn`. Is it ok?